drivers/dai/intel/ssp/ssp.c

```diff
@@ -993,44 +1088,46 @@ static int dai_ssp_bclk_prepare_enable(struct dai_intel_ssp *dp)
 	LOG_INF("sscr0 = 0x%08x", sscr0);
 out:
 	if (!ret) {
-		ssp->clk_active |= SSP_CLK_BCLK_ACTIVE;
+		ssp_link->ssp_plat_data.clk_active |= SSP_CLK_BCLK_ACTIVE;
 	}
 
 	return ret;
 }
 
 static void dai_ssp_bclk_disable_unprepare(struct dai_intel_ssp *dp)
 {
-	struct dai_intel_ssp_pdata *ssp = dai_get_drvdata(dp);
+	struct intel_ssp_link *ssp_link = dai_get_ssp_link(dp);
 
-	if (!(ssp->clk_active & SSP_CLK_BCLK_ACTIVE)) {
+	if (!(ssp_link->ssp_plat_data.clk_active & SSP_CLK_BCLK_ACTIVE)) {
 		return;
 	}
 #if CONFIG_INTEL_MN
-	dai_ssp_mn_release_bclk(dp, dp->index);
+	dai_ssp_mn_release_bclk(dp, ssp_link->link_index);
 #endif
-	ssp->clk_active &= ~SSP_CLK_BCLK_ACTIVE;
+	ssp_link->ssp_plat_data.clk_active &= ~SSP_CLK_BCLK_ACTIVE;
 }
 
 static void dai_ssp_log_ssp_data(struct dai_intel_ssp *dp)
 {
-	LOG_INF("dai index: %u", dp->index);
-	LOG_INF("plat_data base: %u", dp->plat_data.base);
-	LOG_INF("plat_data irq: %u", dp->plat_data.irq);
-	LOG_INF("plat_data fifo playback offset: %u", dp->plat_data.fifo[DAI_DIR_PLAYBACK].offset);
+	LOG_INF("dai index: %u", dp->dai_index);
```

Comment:
Should we display the link number in the logs somehow as it's a new piece of config data?